### PR TITLE
chore: component visual regression tests using percy

### PR DESCRIPTION
**Pull request by Ashley-p44**
Closes #DES-381

### Percy
Added Percy SDK for Storybook and Percy's CLI. 

I ultimately decided this was a way better solution than Cypress + Percy plugin just due to Storybook rendering components within an iframe. TLDR: Cypress does not play nice with iframes. It's very hard to work with the elements in inside iframes and they usually do not show up in test failure snapshots (at least with Cypress). 

I found the the documentation for [Percy Storybook SDK](https://docs.percy.io/docs/storybook) and it is super straight forward and covers all test cases I think we are interested in. If you want additional snapshots for a particular state of a component, just add a new stories in Storybook.   

### CI/CD

- Added the Mainfest PERCY_TOKEN to project secrets
- I added running visual regression tests under CI (this may not be ideal)
   - @bclark-p44 let me know your opinion, we can likely make this a little more deterministic (maybe only run if there is updates in packages or any `.stories.tsx` files)

## Screenshots
image

Example Percy build: https://percy.io/7dd77246/Manifest/builds/23346035

## Merge checklist

- [ ] Added/updated tests
- [ ] Added changeset

### CODE REVIEW

File: .percy.yml [hunk 1]
```diff
@@ -0,0 +1,14 @@
+version: 2
+snapshot:
+  widths:
+    - 1280
```
Comment (Ashley-p44) [inline .percy.yml:4]: @bclark-p44 I limited this to one width to save on screenshots. We can easily add `375px` to cover mobile.

File: packages/react-tooltip/stories/Tooltip.stories.tsx [hunk 2]
```diff
@@ -14,7 +14,7 @@ export default {
 };
 
 const Template: ComponentStory<typeof Tooltip> = (args) => (
-	<Tooltip {...args} title="Tooltip">
+	<Tooltip defaultOpen {...args} title="Tooltip">
```
Comment (Ashley-p44) [inline packages/react-tooltip/stories/Tooltip.stories.tsx:17]: Add default open so tooltip would be present when snapshot was taken.

File: apps/storybook/package.json [hunk 3]
```diff
@@ -5,7 +5,9 @@
   "scripts": {
     "build": "build-storybook -c .storybook -o ./build",
     "dev": "start-storybook -p 6006 --quiet",
-    "clean": "rm -rf .turbo node_modules build"
+    "clean": "rm -rf .turbo node_modules build",
+    "percy": "percy storybook ./build",
```
Comment (Ashley-p44) [inline apps/storybook/package.json:9]: Runs against static Storybook build instead of local or live Storybook URL

Comment (bclark-p44) [reply apps/storybook/package.json:9]: could we hoist this command to the root? would make the calling the script easier.

File: .percy.yml (hunk 1 above)
Comment (bclark-p44) [reply .percy.yml:4]: Yeah we will be adding our adaptive sizes to the design system soon.

File: package.json [hunk 4]
```diff
@@ -14,9 +14,12 @@
     "dev": "turbo run dev",
     "format": "prettier --check \"**/*.{ts,tsx,md}\"",
     "lint": "eslint --ext .tsx,.ts ./packages",
+    "percy": "yarn ./apps/storybook/ exec percy storybook ./build",
     "prepare": "husky install",
     "release-packages": "yarn build && yarn changeset publish",
+    "storybook:build": "yarn ./apps/storybook/ exec build-storybook -c .storybook -o ./build",
```
Comment (bclark-p44) [inline package.json:20]: you should just be able to use a filter with turbo to run this command

`yarn turbo run build --filter=@project44-manifest/storybook`

**Review by bclark-p44 — APPROVED** [review]

**Comment by Ashley-p44** [comment]
@bclark-p44 looks like we ran out of Percy snapshots 😭

**Comment by Ashley-p44** [comment]
Got approval from @bclark-p44 it was okay to merge. CI/CD will fail until snapshots renew in January.

**Pull request merged**